Shared/Services/LoopringGraphQLService.cs

```diff
@@ -1,23 +1,37 @@
 ﻿using Lexplorer.Helpers;
 using Lexplorer.Models;
 using Newtonsoft.Json;
-using Newtonsoft.Json.Converters;
-using Newtonsoft.Json.Serialization;
 using Newtonsoft.Json.Linq;
 using RestSharp;
 using System.Diagnostics;
-using System.Threading.Tasks;
-using System.Collections.Generic;
-using System;
-using System.Linq;
 using RestSharp.Serializers;
-using System.Globalization;
 using System.Text.RegularExpressions;
-using System.Threading;
 
 namespace Lexplorer.Services
 {
-    public class LoopringGraphQLService : IDisposable
+    public interface ILoopringGraphQLService
+    {
+        void Dispose();
+        Task<Account?> GetAccount(string accountId, CancellationToken cancellationToken = default);
+        Task<List<AccountTokenBalance>?> GetAccountBalance(string accountId, CancellationToken cancellationToken = default);
+        Task<IList<AccountNFTSlot>?> GetAccountNFTs(int skip, int first, string accountId, CancellationToken cancellationToken = default);
+        Task<IList<Account>?> GetAccounts(int skip, int first, string? typeName = null, CancellationToken cancellationToken = default);
+        Task<IList<Transaction>?> GetAccountTransactions(int skip, int first, string accountId, double? firstBlockId = null, double? lastBlockId = null, CancellationToken cancellationToken = default);
+        Task<string?> GetAccountTransactionsResponse(int skip, int first, string accountId, double? firstBlockId = null, double? lastBlockId = null, CancellationToken cancellationToken = default);
+        Task<Tuple<double, double>?> GetBlockDateRange(DateTime startDateUTC, DateTime endDateUTC);
+        Task<Block?> GetBlockDetails(int blockId, CancellationToken cancellationToken = default);
+        Task<Blocks?> GetBlocks(int skip, int first, string orderBy = "internalID", string orderDirection = "desc", string? blockTimestamp = null, bool gte = true);
+        Task<NonFungibleToken?> GetNFT(string NFTId, CancellationToken cancellationToken = default);
+        Task<IList<NonFungibleToken>?> GetNFTs(int skip = 0, int first = 10, string orderBy = "mintedAt", string orderDirection = "desc");
+        Task<IList<Transaction>?> GetNFTTransactions(int skip, int first, string nftId, CancellationToken cancellationToken = default);
+        Task<string?> GetNFTTransactionsResponse(int skip, int first, string nftId, CancellationToken cancellationToken = default);
+        Task<Pairs?> GetPairs(int skip = 0, int first = 10, string orderBy = "tradedVolumeToken0Swap", string orderDirection = "desc");
+        Task<Transaction?> GetTransaction(string transactionId);
+        Task<Transactions?> GetTransactions(int skip, int first, string? blockId = null, string? typeName = null, CancellationToken cancellationToken = default);
+        Task<IList<object>?> Search(string searchTerm);
+    }
```

Comment:
What is the point of separating the interface here? I mean I am fully aware of situations where this is important and where you actually gain something from the additional level of abstraction - but why here? We can use the DI just as well without an interface, so I don't see what we gain? The only effect is that we now need to add every new method to the class and the interface.

Reply:
When using DI it is of course possible to use the direct implementation of that class and thereby having one place to add/remove and make changes. With this class with many methods doing actual work it is (for me) easier to have an interface as the contract between the implementation. Not just for seperation for seperation sake, but also to not worry about the actual implementation when working with others. When implemented, the interface in it self is not hard to maintain. When adding to the class you can easily pull up the new methods to the interface, or add to the contract and then add the implementation. 
For testing purposes it also makes it easy to mock ofc. 

If this is not wanted/needed or is just not the way you want it, we can cherry-pick it out of the PR :) I think i made a commit for each. 

As for this specific implementation you could easily change it to another without having to do much in the use of it with an interface. In my experience the real implementation could be switched out without any implications for the application itself.  
But in the end it all comes down to preferences :)

Reply:
Thanks for trying to explain your motives - but to me it still appears as separation for separation's sake. I can't really see that we would mock that interface for testing nor that there are alternative implementations of it. 

And if the interface just serves the purpose of showing a compact form of the provided methods - I find that less of an issue with all the features that a modern IDE offers.

Now speaking of how large that class has become - I also thought about splitting it up in several extension classes. Wouldn't that actually be something that the interface makes impossible?

Reply:
You might be able to make it work, but I don't know why you would split it up since it's implementation is all connected to a single service. It's a lot, I agree, but I personally wouldn't split it up. 

I am not familiar with the service you are using (juanmardefago's repo), but I sence it is not the official loopring graph. Therefore it may be switched out at some point I thought.

Reply:
The official graph (which is actually a fork of juanmardefago's) is broken at the moment and we're pondering about building our own. See #50 

So of course we would do everything to keep it compatible, but yes, that would be one scenario where the implementation would need to be adapted. But again, I don't see us having 2 services around and using one or the other. If that happens, then we surely need to separate that via an interface.